packages/web/src/components/layouts/Error/Error.tsx

```diff
@@ -1,46 +1,161 @@
+        ],
+        numberOfUses: 5265,
+      },
+    },
+    {
+      id: 'TYIwvAfy',
+      title: 'react-smooth',
+      description: 'is a animation library work on React',
+      featuredSites: {
+        iconList: [
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+        ],
+        numberOfUses: 5265,
+      },
+    },
+    {
+      id: 'Lq1pEEX7',
+      title: 'unist-util-position',
+      description: 'unist utility to get the positional info of nodes',
+      featuredSites: {
+        iconList: [
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+        ],
+        numberOfUses: 5265,
+      },
+    },
+    {
+      id: 'cWOgIbmp',
+      title: 'vfile-message',
+      description: 'Create vfile messages',
+      featuredSites: {
+        iconList: [
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+          'https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg',
+        ],
+        numberOfUses: 5265,
+      },
+    },
+    {
+      id: 'UT97Vpoi',
+      title: 'Go to all Popular packages',
+      variant: 'toAll',
+    },
+  ];
+
   return (
     <>
-      <Header />
-      <Section>
-        <h1 className={styles.heading}>{host}</h1>
-        <p className={styles.primary}>{message}</p>
-        <p className={styles.secondary}>{action}</p>
-        <Button onClick={onRetryClick} className={styles.retry}>
-          {actionTitle}
-        </Button>
-        <a
-          href='https://github.com/gradejs/gradejs/issues'
-          target='_blank'
-          rel='noreferrer'
-          onClick={onReportClick}
-        >
-          <Button className={styles.button} variant='action'>
-            Report an issue
-          </Button>
-        </a>
-      </Section>
+      <Header variant='error' />
+
+      <Container>
+        <section className={styles.errorPage}>
+          <div className={styles.textContent}>
+            <p className={styles.host}>{host}</p>
+            <h2 className={styles.heading}>{message}</h2>
+            {action && <p className={styles.desc}>{action}</p>}
+          </div>
+
+          <div className={styles.searchWrapper}>
+            <SearchBar size='large' placeholder={actionTitle} />
+          </div>
+        </section>
+
+        <CardGroups>
+          <CardGroup title='But we have'>
+            <CardList cards={similarCards} />
```

Comment:
nit: Trying to fit separate domain entities within a single component seems like burden. Feels like these `<CardList/>`'s should be separate components.

Reply:
Added as `TODO` for future task/refactor.